### PR TITLE
ARROW-8145: [C++] Rename FileSystem::GetTargetInfos to GetFileInfo

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@nealrichardson I'm sure some R files need regenerating. Could you make the necessary changes in this PR? Thank you.

**Comment by nealrichardson** [comment]
If only there was a GitHub Actions workflow that would just do it for you when you pushed... ;)

Sure, I'll take a look

**Comment by pitrou** [comment]
If only there was an easy-to-remember CLI command that did it for you, locally... ;-)

**Comment by nealrichardson** [comment]
`cd r && make doc && make test` will update everything for you, assuming you've installed the necessary dependencies as described in the README

**Comment by pitrou** [comment]
Ok, R updated some things but I'm not sure those are relevant. Pushed anyway.

**Comment by nealrichardson** [comment]
Yeah those are unrelated but fine. Your global find/replace had already fixed up everything necessary.

**Comment by pitrou** [comment]
I guess I've been lucky then :-)

File: r/DESCRIPTION [hunk 1]
```diff
@@ -37,7 +37,7 @@ Imports:
     utils,
     vctrs
 Roxygen: list(markdown = TRUE, r6 = FALSE, load = "source")
-RoxygenNote: 7.0.2
+RoxygenNote: 7.1.0
```
Comment (bkietz) [inline r/DESCRIPTION]: This seems like it's not relevant to this PR. @nealrichardson is this fine?

**Review by bkietz — APPROVED** [review]

**Comment by wesm** [comment]
> If only there was an easy-to-remember CLI command that did it for you, locally... ;-)

This is also my preference. I've been setting up helper functions to assist with building and testing the R library

https://github.com/wesm/dev-toolchain/blob/master/toolchain/arrow-toolchain.sh#L512

**Pull request closed without merging**